File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E15_DeleteUsingOperationWrapper.scala
package org.judal.examples.scala.jdbc

import org.junit.Test

import org.scalatest.Suite

import org.judal.Using._

import org.judal.storage.scala.IndexableTableOperation

import org.judal.examples.scala.model.Student

/**
 * Delete a student given his internal primary key
 */
class E15_DeleteUsingOperationWrapper extends Suite {

	@Test def demo() = {
		
		E15_DeleteUsingOperationWrapper.setUp

		var op : IndexableTableOperation[Student] = null
		
		using (op) {
			op = new IndexableTableOperation(new Student)
		  op.delete(7)
		}

		E15_DeleteUsingOperationWrapper.tearDown
	}

}

object E15_DeleteUsingOperationWrapper {

	def setUp() = E10_WriteCSVDataIntoTheDatabase.setUp

	def tearDown() = E10_WriteCSVDataIntoTheDatabase.tearDown
  
}

File: aexample/src/main/java/org/judal/examples/java/jdbc/E15_DeleteUsingOperationWrapper.java
package org.judal.examples.java.jdbc;

import org.junit.Test;
import org.judal.examples.java.model.Student;
import org.judal.storage.java.IndexableTableOperation;

/**
 * Delete a student given his internal primary key
 */
public class E15_DeleteUsingOperationWrapper {

	@Test
	public void demo() throws Exception {
		
		setUp();

		try (IndexableTableOperation<Student> op = new IndexableTableOperation<>(new Student())) {
			op.delete(new Integer(7));
		}

		tearDown();
	}

	public static void setUp() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.setUp();
	}

	public static void tearDown() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.tearDown();
	}	
}
